test/unit/commands_test.rb

```diff
@@ -194,3 +194,87 @@ class TestList < HammerCLIForeman::ListCommand
   end
 
 end
+
+describe HammerCLIForeman::ListCommand do
+  class TestList < HammerCLIForeman::ListCommand
+    resource :domains
+    option '--page', 'PAGE', ''
+    option '--per-page', 'PER_PAGE', ''
+  end
+
+  def build_items(cnt)
+    (1..cnt).map do |i|
+      {:id => i, :name => "Item #{i}"}
+    end
+  end
+
+  def expect_paged_call(page, per_page, item_cnt)
+    api_expects(:domains, :index, "List records page #{page}") do |par|
```

Comment:
I was actually hesitating where to put it, but chose unit tests in the end for the reason @mbacovsky mentioned - it's testing behaviour aspect of an abstract command the users don't know about (also without checking what's being printed to the output).
I'm ok to move it if you think it's wrong. In fact, I don't care too much about where the tests are placed as long as they exist ;-)